lib/protocol/protocol.go

```diff
@@ -640,6 +644,16 @@ func (c *rawConnection) send(msg message, done chan struct{}) bool {
 }
 
 func (c *rawConnection) writerLoop() {
+	select {
+	case cc := <-c.clusterConfigBox:
+		err := c.writeMessage(cc)
+		if err != nil {
+			c.internalClose(err)
+			return
+		}
+	case <-c.closed:
+		return
+	}
```

Comment:
What about hello etc?

Reply:
Currently they are done with a function not bound to a connection instance. However I do have a WIP branch (that I totally forgot about) where I want to change that (to do proper error handling in that part too), so it definitely is a concern. Let's put this PR on hold until I figured that out. Thanks!

Reply:
Actually, I think Hello happens before we wrap the connection in a protocol thing, or not?

Reply:
Exactly, and if there is any error we don't send a close message, even after exchange hellos, which isn't optimal (we have long timeouts). So I want to include hellos and do send close messages. So cluster config wont be the first thing sent anymore, it will be hello -> CC -> anything, i.e. this PR isn't sufficient and might just complicate further PRs. As there is no urgent need for this fix, I want to do these two changes together (or at least in a compatible way).

Reply:
We should not do that, as the connection would have to writed to the model to send hello's presumably, or expose some magical method.

The idea is that hello happens so early that there is no way we could accidentially share data.